usl_pipeline/cloud_functions/main.py

```diff
@@ -143,30 +143,53 @@ def wrapper(cloud_event: functions_framework.CloudEvent) -> None:
 def build_and_upload_study_area_chunk(
     cloud_event: functions_framework.CloudEvent,
 ) -> None:
-    """Creates a study area chunk and uploads it to GCS.
+    """Creates a study area chunk, uploads to bucket, and writes to metastore.
 
     This function is triggered when files containing raw geo data for a study area
-    are uploaded to GCS. It will group the files into TAR-files and upload it to a
-    bucket.
+    are uploaded to GCS. It will load the file into study area chunks bucket and
+    write a new study area to metastore if it does not exist.
     """
-    file_name = cloud_event.data["name"]
+    file_name = pathlib.PurePosixPath(cloud_event.data["name"])
     bucket_name = cloud_event.data["bucket"]
 
-    if re.search(file_names.WPS_DOMAIN3_NC_REGEX, file_name):
+    # For AtmoML, only 3rd domain (500m) output files will be used
+    if re.search(file_names.WPS_DOMAIN3_NC_REGEX, file_name.name):
         storage_client = storage.Client()
+        db = firestore.Client()
+
         bucket = storage_client.bucket(bucket_name)
-        file_blob = bucket.blob(file_name)
+        file_blob = bucket.blob(str(file_name))
 
         # For WRF, we can treat each snapshot output file as its own chunk. If
         # multiple snapshots must be processed together, we can choose to group
         # the files together in a TAR before uploading.
-        with file_blob.open("rb") as f:
+        with file_blob.open("rb") as fd:
             study_area_chunk_bucket = storage_client.bucket(
                 cloud_storage.STUDY_AREA_CHUNKS_BUCKET
             )
-            study_area_chunk_bucket.blob(file_name).upload_from_file(f)
+            study_area_chunk_bucket.blob(str(file_name)).upload_from_file(fd)
 
-        # TODO: Write to metastore - create new chunk entry
+            # File names should be in the form <study_area_name>/<file_name>
+            study_area_name = file_name.parent.name
+
+            # Write metastore entry for study area if it does not exist
+            study_area_exists = (
+                metastore.StudyArea.get_ref(db, str(study_area_name)).get().exists
+            )
+            if not study_area_exists:
+                with xarray.open_dataset(fd) as ds:
+                    x_ll_meters = ds.data_vars["XLONG_M"].values[0, 0, 0] * 111
+                    y_ll_meters = ds.data_vars["XLAT_M"].values[0, 0, 0] * 111
+
+                    study_area = metastore.StudyArea(
+                        name=study_area_name,
+                        col_count=200,
+                        row_count=200,
```

Comment:
Is WRF always 200x200? Or is it just 200 for our particular New York study area?

Reply:
Its always going to be 200x200 for all cities